tests/web/test_db_query.py

```diff
@@ -714,3 +720,118 @@ def test_db_query_ignore_consistent() -> None:
     )
     assert result.extra["stats"]["consistent"] is False
     assert result.extra["stats"]["max_threads"] == 10
+
+
+@pytest.mark.redis_db
+def test_db_query_with_disable_lua_scripts() -> None:
+    query, storage, attribution_info = _build_test_query("count(distinct(project_id))")
+    state.set_config("read_through_cache.disable_lua_scripts_sample_rate", 1)
+
+    mock_result = {
+        "data": [{"uniqExact(project_id)": 10001}],
+        "meta": [{"name": "uniqExact(project_id)", "type": "UInt64"}],
+        "profile": {
+            "blocks": 1,
+            "bytes": 64,
+            "elapsed": 0.021712779998779297,
+            "progress_bytes": 0,
+            "rows": 1,
+        },
+    }
+    mock_value = mock_result
+    formatted_query = format_query(query)
+    reader = mock.MagicMock()
+
+    with mock.patch(
+        "snuba.web.db_query.execute_query_with_rate_limits"
+    ) as mock_execute_query:
+        # Set the return value of the mock
+        mock_execute_query.return_value = mock_value
+
+        result = db_query(
+            clickhouse_query=query,
+            query_settings=HTTPQuerySettings(),
+            attribution_info=attribution_info,
+            dataset_name="events",
+            query_metadata_list=[],
+            formatted_query=formatted_query,
+            reader=reader,
+            timer=Timer("foo"),
+            stats={},
+            trace_id="trace_id",
+            robust=False,
+        )
+
+    assert result.extra["stats"]["lua_scripts_disabled"]
+    key = get_query_cache_key(formatted_query)
+    cached_value = _get_cache_partition(reader).get(key)
+    assert cached_value is not None, "cached_value is None"
+    if mock_result["data"] and cached_value.get("data"):
+        mock_data = mock_result.get("data")
+        cached_data = cached_value.get("data")
+        if isinstance(mock_data, list) and isinstance(cached_data, list):
+            if mock_data and cached_data:
+                if mock_data[0] is not None and cached_data[0] is not None:
+                    assert mock_data[0] == cached_data[0]
+    if mock_result["meta"] and cached_value.get("meta"):
+        mock_meta = mock_result.get("meta")
+        cached_meta = cached_value.get("meta")
+        if isinstance(mock_meta, list) and isinstance(cached_meta, list):
+            if mock_meta and cached_meta:
+                if mock_meta[0] is not None and cached_meta[0] is not None:
+                    assert mock_meta[0] == cached_meta[0]
```

Comment:
this test is doing and mocking too much. Why not just ran the same query twice and make sure there's a cache hit?

Reply:
Second test cases handles. I have removed this testcase